falcon/constants.py

```diff
@@ -38,7 +39,16 @@
     if method.strip() != ''
 ]
 
-COMBINED_METHODS = HTTP_METHODS + WEBDAV_METHODS + FALCON_CUSTOM_HTTP_METHODS
+_META_METHODS = [
+    'WEBSOCKET',
```

Comment:
This might have some weird ramifications for WSGI apps... not sure if it is practically significant to warrant a change though.

Consider the following scenario (tested against this branch):
```python
import falcon


class WeirdStuff:

    def on_websocket(self, req, resp):
        resp.media = {'message': 'Hello'}


app = falcon.App()
app.add_route('/test', WeirdStuff())
```
:arrow_right: 
```
$ curl -X WEBSOCKET http://localhost:8000/test
{"message": "Hello"}
```

Reply:
Good catch. I went ahead and explicitly disallowed it just in case.